src/GafferSceneUI/ParameterInspector.cpp

```diff
@@ -108,11 +108,22 @@ Gaffer::ValuePlugPtr ParameterInspector::source( const GafferScene::SceneAlgo::H
 
 	if( auto light = runTimeCast<Light>( sceneNode ) )
 	{
-		if( auto optionalPlug = light->parametersPlug()->getChild<OptionalValuePlug>( m_parameter.name ) )
+		if( m_parameter.shader.string().empty() )
 		{
-			return optionalPlug->enabledPlug()->getValue() ? optionalPlug : nullptr;
+			if( auto optionalPlug = light->parametersPlug()->getChild<OptionalValuePlug>( m_parameter.name ) )
+			{
+				return optionalPlug->enabledPlug()->getValue() ? optionalPlug : nullptr;
+			}
+		}
+		/// \todo Remove the need to search for a `ShaderPlug` by adding such a plug to
+		/// `GafferScene::Light` itself.
+		for( const auto &plug : Plug::Range( *light ) )
+		{
+			if( const auto shaderPlug = runTimeCast<ShaderPlug>( plug ) )
+			{
+				return const_cast<ValuePlug *>( shaderPlug->parameterSource( m_parameter ) );
+			}
 		}
-		return light->parametersPlug()->getChild<ValuePlug>( m_parameter.name );
```

Comment:
Until Lights are guaranteed to provide an internal Shader, we should probably keep this line. That also has the benefit to the reader that we know we're exiting here, and we don't need to scroll down and down to see what's at the end of the function.